static/shared/js/typeahead.js

```diff
@@ -93,39 +94,48 @@ export function get_emoji_matcher(query) {
 
 export function triage(query, objs, get_item = (x) => x) {
     /*
-        We split objs into four groups:
+        We split objs into three groups:
 
             - entire string exact match
-            - match prefix exactly with `query`
-            - match prefix case-insensitively
```

Comment:
Let's retain the match prefix case, for strings that begin with the query. We should retain the case insensitive version (but don't need to specify that in the variable name since everything in this function will be case insensitive) and following it we can have the word boundary matches.